conda_index/index/fs.py

```diff
@@ -0,0 +1,90 @@
+"""
+Minimal (just what conda-index uses) filesystem abstraction.
+
+Allows `fsspec <https://filesystem-spec.readthedocs.io/>`_ to be used to index
+remote repositories, without making it a required dependency.
+"""
+
+from __future__ import annotations
+
+import os
+import os.path
+import typing
+from dataclasses import dataclass
+from numbers import Number
+from pathlib import Path
+
+if typing.TYPE_CHECKING:  # pragma: no cover
```

Comment:
Instead of saying `# pragma: no cover` for all `if TYPE_CHECKING` calls its better to add it as an ignore pattern, see https://github.com/conda/conda/blob/main/pyproject.toml#L103